notifier/main.ts

```diff
@@ -1,12 +1,41 @@
 import * as path from "std/path/mod.ts";
-import { Application } from "oak/mod.ts";
-import { Bot } from "grammy/mod.ts";
+import { Application, Router } from "oak/mod.ts";
+import { Bot, Context, webhookCallback } from "grammy/mod.ts";
+import { FileFlavor, hydrateFiles } from "grammy_files/mod.ts";
 import env from "./env.ts";
 import * as db from "./db.ts";
 import { verifyGitHubWebhook } from "./utils.ts";
 
 const app = new Application();
-const bot = new Bot(env.BOT_TOKEN);
+const router = new Router();
+const bot = new Bot<FileFlavor<Context>>(env.BOT_TOKEN);
+
+bot.api.config.use(hydrateFiles(bot.token));
+
+bot.chatType(["group", "supergroup"]).filter((ctx) =>
+  ctx.chat.id == env.CHAT_ID
+).on("message:document", async (ctx) => {
+  if (
+    !ctx.message.document.file_name?.endsWith(".patch") || !ctx.message.caption
+  ) {
+    return;
+  }
+  const file = await fetch((await ctx.getFile()).getUrl());
```

Comment:
Doesn't this load the entire file in memory?

Reply:
No, it doesn’t, It’s only loading the response status and headers in memory. The actual content is later piped to another request as blob:

https://github.com/grammyjs/website/blob/feat-notifier---add-ability-to-contribute-to-the-website-by-uploading-patch-files/notifier/main.ts#L27

Reply:
Seems like it does: https://developer.mozilla.org/en-US/docs/Web/API/Response/blob

Reply:
Hmm.